packages/front-end/components/Settings/Team/AddOrphanedUserModal.tsx

```diff
@@ -0,0 +1,72 @@
+import { FC, useState } from "react";
+import { useAuth } from "../../../services/auth";
+import Modal from "../../Modal";
+import RoleSelector from "./RoleSelector";
+import { MemberRoleWithProjects } from "back-end/types/organization";
+import UpgradeModal from "../UpgradeModal";
+import useOrgSettings from "../../../hooks/useOrgSettings";
+
+const AddOrphanedUserModal: FC<{
+  mutate: () => void;
+  close: () => void;
+  name: string;
+  email: string;
+  id: string;
+}> = ({ mutate, close, name, email, id }) => {
+  const { defaultRole } = useOrgSettings();
+
+  const [showUpgradeModal, setShowUpgradeModal] = useState(false);
+
+  const [value, setValue] = useState<MemberRoleWithProjects>({
+    role: "admin",
+    limitAccessByEnvironment: false,
+    environments: [],
+    projectRoles: [],
+    ...defaultRole,
+  });
+
+  const { apiCall } = useAuth();
+
+  if (showUpgradeModal) {
+    return (
+      <UpgradeModal
+        close={close}
+        source="add orphaned user"
+        reason={"To enable advanced permissioning,"}
```

Comment:
Is there part of the sentence missing? The trailing comma makes it look like this isn't finished, though I don't know what this part of the UI looks like.